torch/_inductor/lowering.py

```diff
@@ -759,7 +759,7 @@ def nop(x):
     register_lowering(aten.lift_fresh)(nop)
 
 
-@register_lowering(aten.squeeze, type_promotion_kind=None)
+@register_lowering(prims.squeeze, type_promotion_kind=None)
```

Comment:
why this change? Do we have a decomposition that decomposes squeeze into `prim.squeeze`?

Reply:
Well, here is my reasoning, which seems less good when I write it out.

Under several paths, we got to [this condition](https://github.com/pytorch/pytorch/blob/d21f311af880c736b18b5a588583f6162e9abcfa/torch/_inductor/graph.py#L932) and subsequently failed with a `MissingOperatorWithoutDecomp: missing lowering target: prims.squeeze.default`, which I had trouble getting past.

I compared this to the other similar ops which were working and noticed that the other ops whose lowerings were successfully found were in `aten.` whereas this one was in `prim.` so made this change which fixed the failures and didn't seem to create new ones.

I concluded that the lowering was misregistered to `aten.squeeze`, but you're right that the only decomposition I can find is [this one](https://github.com/pytorch/pytorch/blob/d21f311af880c736b18b5a588583f6162e9abcfa/torch/_refs/__init__.py#L4057-L4058).

Reply:
we might need a lowering for `prims.squeeze` if that decomposition is used (not sure if it's regisered or not), but I think we'll still need the lowering for squeeze regardless for the other overloads (those that are not `aten.squeeze.dims`).

Now, when I look at https://github.com/pytorch/pytorch/blob/d21f311af880c736b18b5a588583f6162e9abcfa/torch/_decomp/__init__.py#L427-L428 I see that the ones that are registered are `aten.squeeze.dim` and `default`, but `dims`? It's a bit unclear to me what's the path that's being exercised here. The best way to find that would be to execute `torch.squeeze` and its different overloads with `torch.compile`  and throw around a few breakpoints to see what's executed and what's not.